offset_manager.go

```diff
@@ -369,16 +514,9 @@ func (pom *partitionOffsetManager) NextOffset() (int64, string) {
 }
 
 func (pom *partitionOffsetManager) AsyncClose() {
-	go func() {
-		pom.lock.Lock()
-		defer pom.lock.Unlock()
-
-		for pom.dirty {
```

Comment:
So I think this has actually never worked properly, as you are creating a deadlock here. The state is blocked as long as `pom.dirty`, but the flag will never actually change to `false` nor will `pom.clean` ever fire, because `pom.updateCommitted` will never be able to acquire `pom.lock.Lock()`